js/app.jsx

```diff
@@ -159,6 +187,106 @@ var SearchableRemoteListMixin = {
 
 }
 
+var ListToolbar = React.createClass({
+  render: function() {
+    var newButton = <div></div>;
+    var diffButton = <div></div>;
+
+    var diffURL = "";
+    var diffDisabled = true;
+
+    //create diff button
+    if (this.props.setting.pagename =="caseversion") {
+      if (typeof this.props.checked !== "undefined"){
+        diffURL = "diff.html?lhs=" + this.props.checked[0] + "&rhs=" + this.props.checked[1]
+        if (this.props.checked.length == 2){
+          var diffDisabled = false;
+        }
+      }
+      diffButton = <Button bsStyle="success" target="blank_" href={diffURL} disabled={diffDisabled}>diff</Button>
+    }
+
+    //create new button
+    if (this.props.setting.pagename == "caseversion") {
+      newButton = <Button href='https://moztrap.mozilla.org/manage/case/add/' >+ New Case</Button>
+    } else if (this.props.setting.pagename == "suite") {
+      newButton = <Button bsStyle="success" href='https://moztrap.mozilla.org/manage/suite/add/' >+ New Suite</Button>
+    }
+    return (
+      <Row>
+        <Col md="12">
+          {newButton}
+          {diffButton}
+        </Col>
+      </Row>
+    )
+  }
+})
+
+
+var SearchableList = React.createClass({
+  mixins: [SearchableRemoteListMixin],
+
+  buildURL: function(query) {
+      var api_url =  config.baseUrl + "/api/v1/" + this.props.setting.pagename;
+      var codegen;
+      var pagename = this.props.setting.pagename;
+      if (pagename == "caseversion") {
```

Comment:
I believe we can call the `caseversionCodegen` by `window['caseversionCodegen'](...)`, then we don't need the if anymore. 

```
window[pagename + "Codegen"](..)
```